### ISSUE TITLE
REGEX pattern characters in "execute DBT SQL" causes incorrect SQL to be generated on the target DB

### ISSUE THREAD

**Issue by food-spotter** [innoverio/vscode-dbt-power-user#196]
### Expected behavior

For the following query:

```sql
SELECT t.*
FROM DEV.STAGING.stg_test_table t
WHERE
t.SOME_COLUMN REGEXP '^[0-9]+$'
GROUP BY 1,2
```

I expect the target database (snowflake) to receive and run this query:

```sql
select * from (SELECT t.*
FROM DEV.STAGING.stg_test_table t
WHERE
t.SOME_COLUMN REGEXP '^[0-9]+$'
GROUP BY 1,2) as query limit 500
```

### Actual behavior

The SQL created by the plugin shows up as this in the query history of Snowflake:

```sql
select * from (SELECT t.*
FROM DEV.STAGING.stg_test_table t
WHERE
t.SOME_COLUMN REGEXP '^[0-9]+) as query limit 500
GROUP BY 1,2) as query limit {limit}
```

There is some sort of bug that makes the SQL invalid due to some character(s) in that REGEX string expression of the query. 

### Steps To Reproduce

Run a query against any SQL but use the REGEXP function with the mentioned expression.
```
'^[0-9]+$'
```

### Log output/Screenshots

```json
{
  "code": -1,
  "message": "Runtime Error\n  Database Error\n    001003 (42000): SQL compilation error:\n    parse error line 5 at position 36 near '<EOF>'.\n    syntax error line 4 at position 22 unexpected '^'.\n    syntax error line 4 at position 29 unexpected ')'.\n    syntax error line 5 at position 0 unexpected 'GROUP'.",
  "data": "\"Error: Traceback (most recent call last):\\n  File \\\"/Users/me/de-dbt/.venv/lib/python3.9/site-packages/dbt/adapters/snowflake/connections.py\\\", line 204, in exception_handler\\n    yield\\n  File \\\"/Users/me/de-dbt/.venv/lib/python3.9/site-packages/dbt/adapters/sql/connections.py\\\", line 74, in add_query\\n    cursor.execute(sql, bindings)\\n  File \\\"/Users/me/de-dbt/.venv/lib/python3.9/site-packages/snowflake/connector/cursor.py\\\", line 799, in execute\\n    Error.errorhandler_wrapper(self.connection, self, error_class, errvalue)\\n  File \\\"/Users/me/de-dbt/.venv/lib/python3.9/site-packages/snowflake/connector/errors.py\\\", line 276, in errorhandler_wrapper\\n    handed_over = Error.hand_to_other_handler(\\n  File \\\"/Users/me/de-dbt/.venv/lib/python3.9/site-packages/snowflake/connector/errors.py\\\", line 331, in hand_to_other_handler\\n    cursor.errorhandler(connection, cursor, error_class, error_value)\\n  File \\\"/Users/me/de-dbt/.venv/lib/python3.9/site-packages/snowflake/connector/errors.py\\\", line 210, in default_errorhandler\\n    raise error_class(\\nsnowflake.connector.errors.ProgrammingError: 001003 (42000): SQL compilation error:\\nparse error line 5 at position 36 near '<EOF>'.\\nsyntax error line 4 at position 22 unexpected '^'.\\nsyntax error line 4 at position 29 unexpected ')'.\\nsyntax error line 5 at position 0 unexpected 'GROUP'.\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \\\"/Users/me/.vscode/extensions/innoverio.vscode-dbt-power-user-0.8.5/dist/dbt_integration.py\\\", line 413, in execute_sql\\n    *self.adapter_execute(compiled_sql, fetch=True),\\n  File \\\"/Users/me/.vscode/extensions/innoverio.vscode-dbt-power-user-0.8.5/dist/dbt_integration.py\\\", line 392, in adapter_execute\\n    return self.adapter.execute(sql, auto_begin, fetch)\\n  File \\\"/Users/me/de-dbt/.venv/lib/python3.9/site-packages/dbt/adapters/base/impl.py\\\", line 225, in execute\\n    return self.connections.execute(\\n  File \\\"/Users/me/de-dbt/.venv/lib/python3.9/site-packages/dbt/adapters/sql/connections.py\\\", line 127, in execute\\n    _, cursor = self.add_query(sql, auto_begin)\\n  File \\\"/Users/me/de-dbt/.venv/lib/python3.9/site-packages/dbt/adapters/snowflake/connections.py\\\", line 431, in add_query\\n    connection, cursor = super().add_query(\\n  File \\\"/Users/me/de-dbt/.venv/lib/python3.9/site-packages/dbt/adapters/sql/connections.py\\\", line 83, in add_query\\n    return connection, cursor\\n  File \\\"/usr/local/Cellar/python@3.9/3.9.15/Frameworks/Python.framework/Versions/3.9/lib/python3.9/contextlib.py\\\", line 137, in __exit__\\n    self.gen.throw(typ, value, traceback)\\n  File \\\"/Users/me/de-dbt/.venv/lib/python3.9/site-packages/dbt/adapters/snowflake/connections.py\\\", line 221, in exception_handler\\n    raise DatabaseException(msg)\\ndbt.exceptions.DatabaseException: Database Error\\n  001003 (42000): SQL compilation error:\\n  parse error line 5 at position 36 near '<EOF>'.\\n  syntax error line 4 at position 22 unexpected '^'.\\n  syntax error line 4 at position 29 unexpected ')'.\\n  syntax error line 5 at position 0 unexpected 'GROUP'.\\n\\nDuring handling of the above exception, another exception occurred:\\n\\nTraceback (most recent call last):\\n  File \\\"/Users/me/.vscode/extensions/innoverio.vscode-dbt-power-user-0.8.5/dist/node_python_bridge.py\\\", line 104, in <module>\\n    value = eval(_compile(data['code'], '<input>', 'eval'), _locals)\\n  File \\\"<input>\\\", line 1, in <module>\\n  File \\\"/Users/me/.vscode/extensions/innoverio.vscode-dbt-power-user-0.8.5/dist/dbt_integration.py\\\", line 418, in execute_sql\\n    raise RuntimeException(str(e))\\ndbt.exceptions.RuntimeException: Runtime Error\\n  Database Error\\n    001003 (42000): SQL compilation error:\\n    parse error line 5 at position 36 near '<EOF>'.\\n    syntax error line 4 at position 22 unexpected '^'.\\n    syntax error line 4 at position 29 unexpected ')'.\\n    syntax error line 5 at position 0 unexpected 'GROUP'.\\n\\n\\tat ChildProcess.onMessage (/Users/me/.vscode/extensions/innoverio.vscode-dbt-power-user-0.8.5/dist/extension.js:1991:184)\\n\\tat Object.onceWrapper (node:events:646:26)\\n\\tat ChildProcess.emit (node:events:526:28)\\n\\tat emit (node:internal/child_process:938:14)\\n\\tat process.processTicksAndRejections (node:internal/process/task_queues:84:21)\""
}
```

### Operating System

mac osx, monterey, version 12.6

### dbt version

1.0.8

### dbt Adapter

dbt-snowflake

### dbt Power User version

v0.8.5

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

**Comment by mdesmet** [comment]
We should escape any regex patterns in the query, as we also use regex patterns to remove the limit.

I will release a fix shortly.

**Comment by mdesmet** [comment]
Fixed in 0.8.6
